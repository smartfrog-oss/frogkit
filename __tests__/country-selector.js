import {
  mount,
  shallow
} from 'vue-test-utils'
import CountrySelector from '@/components/country-selector'
import countries from '../data/countries'

const data = {
  countries: countries,
  currentCountry: 'de',
  selectLabel: 'Your Country is missing? Click here:',
  selectPlaceholder: 'Please select a country'
}

const testCountry = {
  "countryCode": "AR",
  "countryName": "Argentinien",
  "currency": "USD",
  "currencySign": "US$",
  "displayName": "Argentina",
  "languages": {
    "en": "Argentina"
  },
  "market": "NHW_2",
  "preferred": true
}

describe('Country Selector component', () => {

  it('Should render component and match snapshot', () => {
    const wrapper = mount(CountrySelector, {
      propsData: data
    })
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('Should emit change event when updateCountry method is called', () => {
    const cmp = shallow(CountrySelector, {
      propsData: data
    })
    const stub = jest.fn()
    cmp.vm.$on('change', stub)
    cmp.vm.updateCountry('AR')
    expect(stub).toBeCalledWith(testCountry)
  })

})
