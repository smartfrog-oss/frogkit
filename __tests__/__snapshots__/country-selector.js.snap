// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Country Selector component Should render component and match snapshot 1`] = `
<div class="fk-countries">
    <flex align="center" justify="space-between" class="fk-countries__toggle"><img src="test-file-stub" alt="Deutschland">
        <p>Deutschland</p>
        <icon icon="angle-down"></icon>
    </flex>
    <!---->
</div>
`;
